build.gradle

```diff
@@ -18,46 +24,85 @@ repositories {
     jcenter()
 }
 
+publishing {
+    publications {
+        customLibrary(MavenPublication) {
+            from components.java
+        }
+    }
+
+    repositories {
+        maven {
+            name = 'mt.edu.um.cf2.jgribx'
+            url = layout.buildDirectory.dir("repo")
+        }
+    }
+}
+
 sourceSets {
-	main {
-		resources.srcDirs = ["src/main/java","src/main/resources"]
-	}
-	test {
-		resources.srcDirs = ["src/test/java", "src/test/resources"]
-	}
+    main {
+        resources.srcDirs = ["src/main/java", "src/main/resources"]
+    }
+    test {
+        resources.srcDirs = ["src/test/java", "src/test/resources"]
+    }
 }
 
 configurations {
     extraLibs
 }
 
 test {
-		testLogging {
-    		events 'started', 'passed'
-		}
+    testLogging {
+        events 'started', 'passed'
+    }
+}
+
+configurations {
+    runtime
+    ajc
+    aspects
+    expandedJars.transitive = false
+    api {
+        extendsFrom(aspects)
+        extendsFrom(expandedJars)
+    }
+    testArtifacts.extendsFrom testRuntime
+    offlinePluginDependency.extendsFrom(compileOnly)
 }
 
 jar {
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
     from {
-        configurations.extraLibs.collect { it.isDirectory() ? it : zipTree(it) }
+        // Expand all the project jars into the main jar so Raptor can find the CDI annotations
+        (configurations.expandedJars).collect { zipTree(it) }
+    }
+    into('lib') {
+        from((configurations.runtimeClasspath - configurations.offlinePluginDependency) - configurations.expandedJars)
+        exclude 'log4j*'
+        exclude 'slf4j*'
+    }
+
+    manifest {
+        attributes 'Main-Class': 'mt.edu.um.cf2.jgribx.CommandLine'
     }
-	manifest {
-		attributes 'Main-Class': 'mt.edu.um.cf2.jgribx.CommandLine'
-	}
 }
 
 // In this section you declare the dependencies for your production and test code
 dependencies {
     extraLibs(
-        [group: 'commons-cli', name: 'commons-cli', version: '1.4'],
-        ['org.slf4j:slf4j-api:1.7.25'],
-        ['edu.ucar:jj2000:5.2']
+            [group: 'commons-cli', name: 'commons-cli', version: '1.4'],
+            ['org.slf4j:slf4j-api:1.7.25'],
+            ['edu.ucar:jj2000:5.2']
     )
     configurations.implementation.extendsFrom(configurations.extraLibs)
 
+    implementation group: 'org.apache.commons', name: 'commons-lang3', version: '3.0'
```

Comment:
Nit: Always nice to use the same group/name/version format across all dependency declarations, i.e. stick with either `group: X, name:Y, version:123` or use the shorthand everywhere (`group:name:version`).